front/src/js/service.js

```diff
@@ -0,0 +1,234 @@
+/**
+ * @author Yannis Malgorn <yannismalgorn@gmail.com>
+ * @see {@link https://github.com/mdl29/scratchy|GitHub}
+ * @requires module:axios/axios
+ */
+/*
+- add axios cdn link in index.html :
+     	<script src="https://unpkg.com/axios@0.21.1/dist/axios.min.js"></script>
+- call service.js in index.html :
+    <script src="/front/js/service.js"></script>
+- start scratchy service : 
+    var api = new ScratchyService("http://localhost:5000/api");
+    api;
+- call a function :
+    api.function();|
+*/
+/**
+ * 
+ * @constructor
+ * @param {string} apiUrl - api url , eg: http://localhost:5000/api
+ */
+function ScratchyService(apiUrl) {
+    let _this = this;
+    this.cacheUsers = []; // List of users, used as cache
+
+//                               ROOM FUNCTIONS
+
+
+    /**
+     *
+     * @async
+     * @augments ScratchyService
+     * @typedef {Object} Room
+     * @param {string} roomID - room id , eg:60895dd62d1a706830c31f10
+     * @param {string} roomTitle - title of a room , eg:my room
+     * @param {string} roomDescription - description of a room, eg:my room description
+     * @returns {Room} - room information , eg:{ oid:"60895dd62d1a706830c31f10" ,title:"example", description : "my description"}
```

Comment:
It doesn't return a room but it returns a Promise<Room>, see the exemple of async functions here : https://jsdoc.app/tags-async.html